kythe/go/extractors/config/smoke/smoke.go

```diff
@@ -38,46 +38,77 @@ import (
 // automatically. The current implementation is to simply check the fraction of
 // files covered by extractor output.
 //
-// TODO(danielmoy): hook up an indexing step and check actual semantic output
-// instead of simple files.
+// Also optionally supports indexing the extractors output.
 //
 // TODO(danielmoy): support more than just java.
-type Tester interface {
-	TestRepo(ctx context.Context, repo string) (Result, error)
-}
+type Tester func(ctx context.Context, repo string) (Result, error)
 
 // Fetcher is a thin wrapper over something which fetches a given repo and
 // writes it to an output directory. Note that the ConfigPath parameter from
 // config.Repo does not affect Fetch at all.
-type Fetcher interface {
-	Fetch(ctx context.Context, repo config.Repo) error
-}
-
-type gitCommandlineFetcher struct{}
+type Fetcher func(ctx context.Context, repo config.Repo) error
 
-func (g gitCommandlineFetcher) Fetch(ctx context.Context, repo config.Repo) error {
+func GitFetch(ctx context.Context, repo config.Repo) error {
 	// TODO(danielmoy): strongly consider go-git instead of os.exec
 	return exec.CommandContext(ctx, "git", "clone", repo.URI, repo.OutputPath).Run()
 }
 
-type harness struct {
-	extractor   config.Extractor
-	configPath  string
-	repoFetcher Fetcher
+// Indexer is a thin wrapper over some kythe indexer which looks for .kindex
+// files in a given inputDir, indexes them, and deposits in outputDir.
+//
+// TODO(danielmoy): kzip?  This is generally a thing that needs supporting in
+// smoke.go and related files now that I think about it.
+type Indexer func(ctx context.Context, inputDir, outputDir string) error
+
+func EmptyIndexer(ctx context.Context, inputDir, outputDir string) error {
+	return nil
 }
 
-// NewGitTestingHarness creates a simple Tester which uses
+type HarnessOptions struct {
```

Comment:
Please document this.

    // HarnessOptions …

Reply:
done